Fix: Add Esc key support to close datepicker and remove focus from input field

mostafa793 opened the pull request:
### What does this PR do?

Adds support for closing the JalaliDatePicker when the **Escape** key is pressed.  
It also removes focus from the input field to allow the datepicker to reopen properly afterward.

---

### Why is this needed?

In Angular applications, pressing the Escape key can typically close modals or dialogs.  
However, when **JalaliDatePicker** was open and Escape was pressed, only the dialog would close —  
while the datepicker remained visible and active. This created a **confusing UX**, where the user  
thought everything was dismissed, but the datepicker was still open and blocked interaction.

---

### Changes made:
- Introduced a `handleEscKey` function to listen for Escape key events
- Registered the keydown listener inside the `show()` method
- On Escape:
  - The input is blurred (`blur()` called)
  - Datepicker is hidden via existing `hide()` method
  - Event listener is removed for cleanup

---

mostafa793 committed:
Fix: Add Esc key support to close datepicker and remove focus from input field

majidh1 commented on src/index.js:
```diff
@@ -402,6 +404,14 @@ function addEventListenerOnBody() {
 
+function handleEscKey(event) {
+    if (event.key === "Escape") {
+        document.removeEventListener("keydown", handleEscKey);
```
این خط کد در متد hide نیز وجود دارد با توجه به فراخوانی این متد در خط پایین نیازی به تکرار در اینجا نیست

mostafa793 replied:
بله، انجام شد.

majidh1 commented on src/index.js:
```diff
@@ -402,6 +404,14 @@ function addEventListenerOnBody() {
+function handleEscKey(event) {
+    if (event.key === "Escape") {
+        document.removeEventListener("keydown", handleEscKey);
+        jalaliDatepicker.input.blur();
```
احتمال عدم وجود input وجود دارد به همین دلیل بهتر هست define بودن input بررسی شود
```js
jalaliDatepicker.input?.blur?.();
 ```

mostafa793 replied:
درسته، انجام شد.

majidh1 commented on src/index.js:
```diff
@@ -150,12 +150,14 @@ const jalaliDatepicker = {
         }, 300);
         this.setPosition();
         setScrollOnParent(input);
+        document.addEventListener("keydown", handleEscKey);
```
برای هماهنگی بیشتر با ساختار کد از constant برای keydown استفاده شود مثل EVENT_FOCUS_STR

mostafa793 replied:
این مورد هم انجام شد.

majidh1 reviewed (commented):
با تشکر از افزودن این ویژگی خوب لطفا موارد کامنت شده رو بررسی کنید

mostafa793 committed:
Refactor escape key handling and improve show/hide transition behavior

- Removed redundant removeEventListener("keydown", handleEscKey) from handleEscKey, as it's already handled in hide()
- Used optional chaining for input.blur() to avoid potential undefined error
- Replaced hardcoded "keydown" with EVENT_KEYDOWN_STR constant for consistency
- Added isTransitioning flag to prevent closing the datepicker during transition when pressing Escape

mostafa793 commented:
تغییرات طبق بازخورد اعمال شد ✔️
یک موری هم نیاز به اصلاح داشت که انجام شد.
